web/modules/weather_data/src/Service/WeatherDataService.php

```diff
@@ -657,6 +657,57 @@ public function getCurrentConditionsFromGrid($wfo, $gridX, $gridY)
         ];
     }
 
+    public function getHourlyPrecipitation(
+        $wfo,
+        $gridX,
+        $gridY,
+        $now = false,
+        $self = false,
+    ) {
+        date_default_timezone_set("America/New_York");
+
+        if (!$self) {
+            $self = $this;
+        }
+
+        $wfo = strtoupper($wfo);
+        if (!($now instanceof \DateTimeImmutable)) {
+            $now = new \DateTimeImmutable();
+        }
+
+        $place = $self->getPlaceFromGrid($wfo, $gridX, $gridY);
+        $timezone = $place->timezone;
+
+        $forecast = $self->getFromWeatherAPI("/gridpoints/$wfo/$gridX,$gridY")
```

Comment:
Is this the only place where we currently call this API route? If not, we should consider adding a stashed variable on the data service instance and checking that first

Reply:
It's not the only place. I think we use this particular endpoint a few times. Good call on pulling it out and storing it in-memory until the end of the response. I think that will also fit well with the testing changes I'm proposing, as it will help delineate a "data layer" vs. a processing layer.